vscode-trace-webviews/src/common/vscode-message-manager.ts

```diff
@@ -40,9 +39,15 @@ export class VsCodeMessageManager extends Messages.MessageManager {
         vscode.postMessage({command: 'webviewReady'});
     }
 
+    notifyConnection(serverStatus: boolean): void {
+        const status: string = JSON.stringify(serverStatus);
```

Comment:
You don't need to do JSON.stringify() and can remove it. The vscode postMessage() will do JSON.stringify() when sending and JSON.parse() when receiving the message. 

The reason why below JSONBig.stringify() is called is that the data structure that is sent contains BigInt and JSON.stringify() of postMessage() will cause and exception because the standard JSON.stringify() can't handle such special types (BigInt). So, in our code we do  JSONBig.stringify() to convert it to string, which then in postMessage() JSON.stringify() is called and on the receiving side, after JSON.parse() the trace extension has special code to recreate the data structure with BigInts

Reply:
> You don't need to do JSON.stringify() and can remove it. The vscode postMessage() will do JSON.stringify() when sending and JSON.parse() when receiving the message.

Not necessarily, according to my local testing of both approaches. Here, posting this boolean without explicitly stringifying it leads to no status received on the other end (hence no rendering). This is what I briefly mention (why this) in the commit message and PR description. This is one of the few unexpected behaviours I had to figure out recently indeed.

> The reason why below JSONBig.stringify() is called is that the data structure that is sent contains BigInt and JSON.stringify() of postMessage() will cause and exception because the standard JSON.stringify() can't handle such special types (BigInt). So, in our code we do JSONBig.stringify() to convert it to string, which then in postMessage() JSON.stringify() is called and on the receiving side, after JSON.parse() the trace extension has special code to recreate the data structure with BigInts

Yes, I understood that but thanks.

Reply:
You can remove it, it's not needed. Here do: 
```vscode.postMessage({command: 'connectionStatus', data: { serverStatus }});```

The receiving side needs to be updated as well. I'll put comment at one place on how.

Reply:
@marco-miller Your implementation works on my side too. So, since it seems to be a different behaviour without the extra marshalling, let's then merge it as is.